common/functions/src/scalars/maths/pi.rs

```diff
@@ -15,47 +15,44 @@
 use std::f64::consts::PI;
 use std::fmt;
 
-use common_datavalues::prelude::*;
-use common_datavalues::DataTypeAndNullable;
+use common_datavalues2::prelude::*;
 use common_exception::Result;
 
-use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function2_factory::Function2Description;
 use crate::scalars::function_factory::FunctionFeatures;
-use crate::scalars::Function;
+use crate::scalars::Function2;
 
 #[derive(Clone)]
 pub struct PiFunction {
     display_name: String,
 }
 
 impl PiFunction {
-    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function2>> {
         Ok(Box::new(PiFunction {
             display_name: display_name.to_string(),
         }))
     }
 
-    pub fn desc() -> FunctionDescription {
-        FunctionDescription::creator(Box::new(Self::try_create))
+    pub fn desc() -> Function2Description {
+        Function2Description::creator(Box::new(Self::try_create))
             .features(FunctionFeatures::default().deterministic())
     }
 }
 
-impl Function for PiFunction {
+impl Function2 for PiFunction {
     fn name(&self) -> &str {
         &*self.display_name
     }
 
-    fn return_type(&self, _args: &[DataTypeAndNullable]) -> Result<DataTypeAndNullable> {
-        let dt = DataType::Float64;
-        Ok(DataTypeAndNullable::create(&dt, false))
+    fn return_type(&self, _args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        Ok(Float64Type::arc())
     }
 
-    fn eval(&self, _columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
-        Ok(DataColumn::Constant(
-            DataValue::Float64(Some(PI)),
-            input_rows,
-        ))
+    fn eval(&self, _columns: &ColumnsWithField, input_rows: usize) -> Result<ColumnRef> {
+        let mut builder: ColumnBuilder<f64> = ColumnBuilder::with_capacity(1);
+        builder.append(PI);
+        Ok(builder.build(1))
```

Comment:
It looks like this will be fine, what do you think? cc @sundy-li

Reply:
I think
```
ConstColumn::new(Series::from_data(vec![PT]), input_rows)
```
will be fine.

Reply:
but if change it like you said, we will get

```
Diff < left / right > :
<PrimitiveColumn 	 typeid: Float64	 len: 1	 data: [3.141592653589793]
>PrimitiveColumn 	 typeid: Float64	 len: 0	 data: []
```

Reply:
Oh, there is a bug in:

```
        with_match_primitive_type_id!(dt, |$T| {
            fmt_dyn!(&self, PrimitiveColumn<$T>, f)
        },
```
change &self to `col` will fix this, it's ok to fix this in this pr.

Reply:
got it